### ISSUE TITLE
无法复现结果

### ISSUE THREAD

**Issue by cnexah** [Banconxuan/RTM3D#36]
您好！我使用预训练的model_res18_2.pth，利用src/demo.py进行预测。python3.7, pytorch1.8.1, cuda10.2
命令：
python src/demo.py --demo kitti_format/data/kitti/val.txt --data_dir kitti_format --calib_dir kitti_format/data/kitti/calib/ --load_model model_res18_2.pth  --gpus 0 --arch res_18
输出：
NMS not imported! If you need it, do 
 cd $CenterNet_ROOT/src/lib/external 
 make
Fix size testing.
training chunk_sizes: [32]
The output will be saved to  kitti_format/exp/default
heads {'hm': 3, 'wh': 2, 'hps': 18, 'rot': 8, 'dim': 3, 'prob': 1, 'reg': 2, 'hm_hp': 9, 'hp_offset': 2}
Creating model...
=> loading pretrained model https://download.pytorch.org/models/resnet18-5c106cde.pth
model_res18_2.pth
loaded model_res18_2.pth, epoch 199
results_dir kitti_format/exp/results
000554 [2.552675724029541, 1.5896803140640259, -0.03506125509738922]
004822 [2.438274621963501, 1.3753342628479004, -0.11458498984575272]
005190 [2.5769827365875244, 1.5554676055908203, -0.037575557827949524]
005695 [2.445730447769165, 1.4386816024780273, -0.007032535970211029]

测评得到car的结果是这样的：
Car AP@0.70, 0.70, 0.70:
bbox AP:87.68, 78.48, 69.47
bev  AP:21.21, 16.34, 15.16
3d   AP:16.47, 13.53, 12.04
aos  AP:86.76, 77.13, 68.13
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:90.74, 82.73, 71.02
bev  AP:15.26, 10.09, 8.57
3d   AP:9.70, 6.33, 5.24
aos  AP:89.70, 81.12, 69.46

请问我输入的命令是有问题的吗？非常感谢！

**Comment by GUUUUG** [comment]
我也无法预测得到他给出的效果

**Comment by DuZzzs** [comment]
同样无法复现结果。
- 使用readme中第一个预训练模型res18，kitti-split1，进行测试。（由于部分图网络没有检测到目标，在eval时会报错（你们没遇到？？）修改代码不算入这些图，理论上下面计算出的指标应该更高些）。测试结果如下，低于readme中列出的指标：
```
Car AP@0.70, 0.70, 0.70:
bbox AP:90.02, 79.79, 70.61
bev  AP:23.77, 17.18, 16.02
3d   AP:17.79, 13.85, 12.25
aos  AP:89.06, 78.50, 69.24
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:93.30, 84.25, 74.34
bev  AP:18.56, 12.08, 9.71
3d   AP:11.61, 7.07, 5.25
aos  AP:92.25, 82.70, 72.65
Car AP@0.70, 0.50, 0.50:
bbox AP:90.02, 79.79, 70.61
bev  AP:52.52, 36.88, 33.66
3d   AP:48.28, 34.12, 28.21
aos  AP:89.06, 78.50, 69.24
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:93.30, 84.25, 74.34
bev  AP:52.65, 35.48, 30.41
3d   AP:45.86, 31.05, 25.55
aos  AP:92.25, 82.70, 72.65
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:61.10, 52.19, 43.72
bev  AP:10.22, 10.28, 10.25
3d   AP:10.03, 10.22, 9.09
aos  AP:50.87, 43.29, 36.39
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:63.76, 52.08, 44.73
bev  AP:2.34, 2.04, 1.84
3d   AP:1.76, 1.78, 1.30
aos  AP:51.80, 41.95, 35.77
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:61.10, 52.19, 43.72
bev  AP:20.10, 17.29, 16.39
3d   AP:19.67, 16.89, 16.26
aos  AP:50.87, 43.29, 36.39
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:63.76, 52.08, 44.73
bev  AP:13.46, 11.51, 9.33
3d   AP:13.04, 11.17, 9.12
aos  AP:51.80, 41.95, 35.77
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:54.79, 31.88, 31.23
bev  AP:14.50, 10.97, 10.97
3d   AP:12.58, 11.20, 11.20
aos  AP:51.17, 29.87, 29.19
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:53.33, 30.53, 28.47
bev  AP:7.10, 3.33, 3.40
3d   AP:6.26, 3.30, 3.17
aos  AP:49.16, 27.98, 25.90
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:54.79, 31.88, 31.23
bev  AP:21.18, 13.64, 13.67
3d   AP:21.12, 13.64, 13.67
aos  AP:51.17, 29.87, 29.19
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:53.33, 30.53, 28.47
bev  AP:15.21, 7.92, 7.59
3d   AP:15.15, 7.87, 7.50
aos  AP:49.16, 27.98, 25.90
```
- 自己重新训练了res18，在val上进行测试，结果远低于readme中列出的指标。
`python ./src/main.py --data_dir /opendata/kitti/object_detect/kitti_split1_rtm3d \
--exp_id km3d_multi_class --arch res_18 --gpus 0 --batch_size 8 --lr 1.25e-4 --num_epochs 200 --save_all`
```
Car AP@0.70, 0.70, 0.70:
bbox AP:90.24, 80.06, 70.84
bev  AP:0.08, 0.08, 0.09
3d   AP:0.03, 0.03, 0.03
aos  AP:89.53, 78.99, 69.64
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:95.46, 84.54, 74.52
bev  AP:0.02, 0.02, 0.02
3d   AP:0.00, 0.00, 0.01
aos  AP:94.60, 83.23, 73.02
Car AP@0.70, 0.50, 0.50:
bbox AP:90.24, 80.06, 70.84
bev  AP:47.54, 34.28, 28.15
3d   AP:37.91, 26.42, 23.41
aos  AP:89.53, 78.99, 69.64
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:95.46, 84.54, 74.52
bev  AP:45.69, 31.03, 25.45
3d   AP:35.48, 23.01, 18.30
aos  AP:94.60, 83.23, 73.02
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:61.33, 52.53, 43.87
bev  AP:9.09, 9.09, 9.09
3d   AP:3.03, 3.03, 3.03
aos  AP:52.22, 44.35, 37.09
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:64.06, 54.48, 45.10
bev  AP:0.96, 0.69, 0.60
3d   AP:0.49, 0.42, 0.42
aos  AP:53.18, 44.51, 36.74
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:61.33, 52.53, 43.87
bev  AP:16.00, 15.54, 12.92
3d   AP:15.43, 15.22, 12.78
aos  AP:52.22, 44.35, 37.09
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:64.06, 54.48, 45.10
bev  AP:9.96, 8.47, 7.20
3d   AP:9.34, 7.95, 6.81
aos  AP:53.18, 44.51, 36.74
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:52.53, 35.18, 30.56
bev  AP:0.52, 0.52, 0.52
3d   AP:0.35, 0.35, 0.35
aos  AP:48.51, 32.44, 28.46
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:51.82, 30.79, 29.03
bev  AP:0.29, 0.14, 0.14
3d   AP:0.09, 0.09, 0.09
aos  AP:47.14, 27.70, 26.19
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:52.53, 35.18, 30.56
bev  AP:15.73, 10.39, 8.62
3d   AP:15.07, 8.03, 8.08
aos  AP:48.51, 32.44, 28.46
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:51.82, 30.79, 29.03
bev  AP:14.19, 7.79, 7.11
3d   AP:13.02, 6.79, 6.74
aos  AP:47.14, 27.70, 26.19
```
有人遇到这个问题并解决的吗？？

**Comment by GUUUUG** [comment]
me too

**Comment by wangziniu1109** [comment]
> * 下
请问一下你eval的时候报的什么错啊，还有能告诉我你是怎么修改的代码吗

**Comment by GUUUUG** [comment]
我没有报错啊




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年7月1日(星期四) 晚上9:40
收件人: ***@***.***&gt;; 
抄送: "王广杰"<953704875@QQ.COM&gt;; ***@***.***&gt;; 
主题: Re: [Banconxuan/RTM3D] 无法复现结果 (#36)





   
下
 请问一下你eval的时候报的什么错啊，还有能告诉我你是怎么修改的代码吗

**Comment by tfwang08** [comment]
请问现在复现出来了吗？我还不行。

**Comment by GUUUUG** [comment]
我也是，差他的指标一两个点




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年7月4日(星期天) 晚上10:35
收件人: ***@***.***&gt;; 
抄送: "王广杰"<953704875@QQ.COM&gt;; ***@***.***&gt;; 
主题: Re: [Banconxuan/RTM3D] 无法复现结果 (#36)





 
请问现在复现出来了吗？我还不行。

**Comment by tfwang96** [comment]
> 我也是，差他的指标一两个点
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2021年7月4日(星期天) 晚上10:35 收件人: ***@***.***&gt;; 抄送: "王广杰"<953704875@QQ.COM&gt;; ***@***.***&gt;; 主题: Re: [Banconxuan/RTM3D] 无法复现结果 (#36) 请问现在复现出来了吗？我还不行。 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

那还挺奇怪的，可能不是我们的问题吧。

**Comment by wufeim** [comment]
大无语...
